src/luascript.cpp

```diff
@@ -4256,6 +4338,33 @@ int LuaScriptInterface::luaSendGuildChannelMessage(lua_State* L)
 	return 1;
 }
 
+
+int LuaScriptInterface::luaLoadDirectory(lua_State* L)
+{
+	const boost::filesystem::path& dir = boost::filesystem::current_path().string() + getString(L, 1);
```

Comment:
You can use auto (to make it more readable) and we potentially do less work by letting the implementation of path do the string operation.